src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-form/smb-cluster-form.component.html

```diff
@@ -0,0 +1,372 @@
+        <option value="">-- Select the auth mode --</option>
+        <option value="active-directory">Active Directory</option>
+        <option value="user">User</option>
+      </cds-select>
+
+      <ng-template #authModeError>
+        <span
+          class="invalid-feedback"
+          *ngIf="smbForm.showError('auth_mode', formDir, 'required')"
+          i18n
+          >This field is required.</span
+        >
+      </ng-template>
+    </div>
+
+    <!-- Domain Setting -->
+    <div class="form-item"
+         *ngIf="this.smbForm.get('auth_mode').value === 'active-directory'">
+      <div cdsCol
+           [columnNumbers]="{ md: 12 }"
+           class="d-flex">
+        <cds-text-label
+          for="domain_setting"
+          i18n
+          cdRequiredField="Domain Setting"
+          [invalid]="
+            smbForm.controls.domain_setting.invalid && smbForm.controls.domain_setting.dirty
+          "
+          [invalidText]="domainError"
+          >Domain Setting
+          <div class="cds-input-group">
+            <input
+              cdsText
+              type="text"
+              placeholder="Domain Setting..."
+              i18n-placeholder
+              id="domain_setting"
+              name="domain_setting"
+              formControlName="domain_setting"
+              [invalid]="
+                smbForm.controls.domain_setting.invalid && smbForm.controls.domain_setting.dirty
+              "
+            />
+
+            <cds-icon-button kind="ghost"
+                             size="md">
+              <svg cdsIcon="edit"
+                   size="32"
+                   class="cds--btn__icon"
+                   icon></svg>
+            </cds-icon-button>
+
+            <cds-icon-button kind="ghost"
+                             size="md">
+              <svg cdsIcon="trash-can"
+                   size="32"
+                   class="cds--btn__icon"
+                   icon></svg>
+            </cds-icon-button>
+            <button cdsButton="tertiary"
+                    type="button"
+                    (click)="showDomainSettingModal()">
+              Domain Setting
+              <svg cdsIcon="add"
+                   size="32"
+                   class="cds--btn__icon"
+                   icon></svg>
+            </button>
+          </div>
+        </cds-text-label>
+      </div>
+    </div>
+
+    <!-- User Group Setting -->
+    <div class="form-item"
+         *ngIf="this.smbForm.get('auth_mode').value === 'user'">
+      <cds-text-label for="user_group_setting"
+                      i18n
+                      cdRequiredField="User Group Setting">User Group Setting
+        <input
+          cdsText
+          type="text"
+          placeholder="User Group Setting..."
+          i18n-placeholder
+          id="user_group_setting"
+          name="user_group_setting"
+          formControlName="user_group_setting"
+        />
+      </cds-text-label>
+    </div>
+
+    <!-- Joint Source -->
```

Comment:
I see the word `Joint` here and a bunch of spots that follow. *If* you mean `join` as in "join to active directory" I suggest correcting this to `join`, `joinSources`, etc.  If `joint` means something to the GUI framework that I'm unaware of, feel free to ignore this message. 

(PS. Even though the words are similar - we `join` the club, my elbow is a `joint` (and many other slang meanings ;-))